server/app/views/admin/programs/ProgramBlockBaseView.java

```diff
@@ -0,0 +1,108 @@
+package views.admin.programs;
+
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.li;
+import static j2html.TagCreator.span;
+import static j2html.TagCreator.text;
+import static j2html.TagCreator.ul;
+
+import com.google.common.collect.ImmutableList;
+import j2html.tags.specialized.ButtonTag;
+import j2html.tags.specialized.DivTag;
+import j2html.tags.specialized.UlTag;
+import services.program.ProgramDefinition;
+import services.program.predicate.PredicateDefinition;
+import services.program.predicate.PredicateExpressionNode;
+import services.question.types.QuestionDefinition;
+import views.BaseHtmlView;
+import views.ViewUtils;
+import views.ViewUtils.ProgramDisplayType;
+import views.components.Icons;
+import views.style.AdminStyles;
+
+abstract class ProgramBlockBaseView extends BaseHtmlView {
+
+  /** Renders a div with internal/admin program information. */
```

Comment:
**nitpick (blocking):** Would you mind fleshing out what is rendered more?  Our View javadocs are fairly lacking imo so it's hard to know what is visually being created in methods.

So I think here it'd include rendering the banner with program details

Reply:
Please see other comment about cleanups tracked in #4019